bionic/flow.py

```diff
@@ -694,6 +694,13 @@ def __call__(self, func_or_provider):
             provider = DEFAULT_PROTOCOL(provider)
         if provider.attrs.should_persist is None:
             provider = decorators.persist(True)(provider)
+        if provider.attrs.should_memoize is None:
+            provider = decorators.memoize(True)(provider)
+        if not (provider.attrs.should_persist or provider.attrs.should_memoize):
+            raise ValueError(
+                "Attempted to set both persist and memoize to False. "
+                "Must store computed entity for downstream computation. "
```

Comment:
We should probably mention the name of the entity here to help with debugging.

Also, I wonder if the second sentence is a bit cryptic?  Maybe something like "At least one form of storage must be enabled."?